File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDOLostFindParamModel.kt
//
//  IDOLostFindParamModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.GsonBuilder
import com.google.gson.annotations.SerializedName

///
/// Set Lost Find Event

open class IDOLostFindParamModel(mode: Int) : IDOBaseModel {

    /// Mode
    /// 0: No anti-lost
    /// 1: Close-range anti-lost
    /// 2: Medium-range anti-lost
    /// 3: Long-range anti-lost 
    @SerializedName("mode")
    var mode: Int = mode


    

    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }
}
    

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDOLostFindParamModel.swift

//
//  IDOLostFindParamModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/12.
//

import Foundation
/// Set Lost Find Event
@objcMembers
public class IDOLostFindParamModel: NSObject, IDOBaseModel {
    /// Mode
    /// 0: No anti-lost
    /// 1: Close-range anti-lost
    /// 2: Medium-range anti-lost
    /// 3: Long-range anti-lost 
    public var mode: Int
    
    enum CodingKeys: String, CodingKey {
        case mode
    }
    
    public init(mode: Int) {
        self.mode = mode
    }
    
    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}

